### PR TITLE
Functional tests: Map Boundaries

### PR DESCRIPTION
**Pull request by OlafSzmidt**

### CODE REVIEW

File: aimmo-game/tests/functional/test_map_movements.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
+from unittest import TestCase
+
+from mock_world import MockWorld
+from simulation.location import Location
+from simulation import map_generator
+from tests.test_simulation.dummy_avatar import MoveEastDummy, MoveWestDummy, MoveNorthDummy, MoveSouthDummy, WaitDummy
+
+
+class TestMapMovements(TestCase):
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_map_movements.py]: Are you testing the Map Movements, or the movements within the map? :thinking:

File: aimmo-game/tests/functional/test_map_movements.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+    SETTINGS = {
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_map_movements.py]: somehow inconsistent, in test_action.py, the settings are not in a SETTINGS dictionary. Maybe open an issue for this later?

File: aimmo-game/tests/test_simulation/dummy_avatar.py [hunk 3]
```diff
@@ -39,17 +39,17 @@ def damage(self, amount):
```
Comment (CelineBoudier) [inline aimmo-game/tests/test_simulation/dummy_avatar.py]: I found weird stuff in dummy_avatar, I'll open issues for them as it's not fully related to that PR

File: aimmo-game/tests/test_simulation/test_map_generator.py [hunk 4]
```diff
@@ -92,7 +92,7 @@ def test_map_dimensions(self):
                 self.assertGreaterEqual(c.location.x, -1)
                 self.assertGreaterEqual(c.location.y, -1)
 
-    def test_obstable_ratio(self):
+    def test_obstacle_ratio(self):
```
Comment (CelineBoudier) [inline aimmo-game/tests/test_simulation/test_map_generator.py]: :joy:

File: aimmo-game/tests/functional/test_map_movements.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+    def set_up_environment(self, dummy_list=None, location=Location(0, 0), map_generator_class=map_generator.Main):
+        """
+        Utility method for testing.
+        """
+        self.game = MockWorld(TestMapMovements.SETTINGS, dummy_list, map_generator_class)
+        self.game.game_state.add_avatar(1, None, location)
+        self.avatar = self.game.avatar_manager.get_avatar(1)
+
+    def test_movement_five_times_in_all_directions(self):
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_map_movements.py]: can't all these bits be more factorised somehow? Seems  a bit clunky. Also when you write "consists of 5 tests" or "4 tests", you sort of expect them to be in different functions?

File: aimmo-game/tests/test_simulation/dummy_avatar.py (hunk 3 above)
Comment (CelineBoudier) [reply aimmo-game/tests/test_simulation/dummy_avatar.py]: see #353

File: aimmo-game/tests/functional/test_map_movements.py (hunk 1 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/functional/test_map_movements.py]: :+1:

File: aimmo-game/tests/functional/test_map_movements.py (hunk 2 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/functional/test_map_movements.py]: There are no settings in `test_action.py`? If you mean the constants at the top such as `ORIGIN` or `SOUTH_OF_ORIGIN` then I personally think it's a different case scenario;

`SETTINGS` in `MockWorld` needs to be supplied as a whole dictionary and is used to update the existing dictionary in the World, these values are just constants used on their own.

I can put them into a dictionary and access them individually from there (maybe from an issue); as you wish ;)

File: aimmo-game/tests/functional/test_map_movements.py (hunk 5 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/functional/test_map_movements.py]: I used to put stuff like this in separate tests but in a previous PR was told by Dips and Duncan not to do so if I'm only changing certain values but not the actual edge case. I think here we're testing only one edge case in 4 different scenario's which is why I left it inside one test.

I will change the comment though! :+1:

File: aimmo-game/tests/test_simulation/dummy_avatar.py (hunk 3 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/test_simulation/dummy_avatar.py]: :+1:

File: aimmo-game/tests/test_simulation/test_map_generator.py (hunk 4 above)
Comment (OlafSzmidt) [reply aimmo-game/tests/test_simulation/test_map_generator.py]: XDDDDD

File: aimmo-game/tests/functional/test_movements_in_map.py [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+        # West.
+        self.set_up_environment([MoveWestDummy])
+        self.assertEqual(self.avatar.location, Location(0, 0))
+
+        for i in range(5):
+            self.game.turn_manager._run_single_turn()
+
+        self.assertEqual(self.avatar.location, Location(-5, 0))
+
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_movements_in_map.py]: it's just that it looks like it could easily be templated, instead of writing the same code over and over again?

File: aimmo-game/tests/functional/test_movements_in_map.py [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    def set_up_environment(self, dummy_list=None, location=Location(0, 0), map_generator_class=map_generator.Main):
+        """
+        Utility method for testing.
+        """
+        self.game = MockWorld(TestMovementsInMap.SETTINGS, dummy_list, map_generator_class)
+        self.game.game_state.add_avatar(1, None, location)
+        self.avatar = self.game.avatar_manager.get_avatar(1)
+
+    def set_up_and_make_movements_in_a_single_direction(self, dummy_list, no_of_movements, spawn=Location(0, 0)):
```
Comment (CelineBoudier) [inline aimmo-game/tests/functional/test_movements_in_map.py]: i'd use nbr_of_ number_of, or even nr_of but not no_of as it sound stoo much like a negation
